uaclient/api/u/pro/security/fix/__init__.py

```diff
@@ -318,9 +425,18 @@ def register_step(
                 order=self.order, data=EnableData.from_dict(data)
             )
         elif operation == FixStepType.NOOP:
-            fix_step = FixPlanNoOpStep(
-                order=self.order, data=NoOpData.from_dict(data)
-            )
+            if "patch_version" in data:
+                fix_step = FixPlanNoOpLivepatchFixStep(
+                    order=self.order, data=NoOpLivepatchFixData.from_dict(data)
+                )
+            elif "source_packages" in data:
+                fix_step = FixPlanNoOpAlreadyFixedStep(
+                    order=self.order, data=NoOpAlreadyFixedData.from_dict(data)
+                )
+            else:
+                fix_step = FixPlanNoOpStep(
+                    order=self.order, data=NoOpData.from_dict(data)
+                )
```

Comment:
I like how the inheritance works here, and I don't think we need a version bump to accommodate this change. The other changes being just additions, I'd say we good overall.